test/dynamo/test_misc.py

```diff
@@ -12096,6 +12096,24 @@ def run(x, s, foo0):
         self.assertTrue(same(res[0], torch.ones(1) + 1))
         self.assertEqual(res[1], 1)
 
+    def test_ne_operator_with_custom_eq(self):
+        class Foo:
+            def __init__(self, x):
+                self.x = x
+
+            def __eq__(self, other):
+                return self.x == other.x
+
+        @torch.compile(fullgraph=True, backend="eager")
+        def run(x):
+            f1 = Foo(0)
+            f2 = Foo(0)
+            # `x + 1` prevents Dynamo from skipping this frame.
+            return x + 1, f1 != f2
+
+        _, ne = run(torch.ones(1))
+        self.assertFalse(ne)
+
```

Comment:
Can you add a test for a user-defined `__eq__`? It would be cool to test one that has a graph break in it as well (just to check that dynamo isn't blindly executing the `__eq__`.

Reply:
> Can you add a test for a user-defined __eq__?

Hmm isn't this test just that? `Foo.__eq__` is user defined.

> It would be cool to test one that has a graph break in it as well (just to check that dynamo isn't blindly executing the __eq__

You mean `torch._dynamo.graph_break()` inside the `__eq__`, and check `torch.compile` did induce a graph break?

Reply:
> Hmm isn't this test just that? Foo.__eq__ is user defined.

Whoops I can't read

> You mean torch._dynamo.graph_break() inside the __eq__, and check torch.compile did induce a graph break?

Yup

Reply:
Cool, added!